File: front/src/app/core/commands/CheckResetCodeCommand.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ServerMessage } from '../services/ServerMessageService';
import { CommandHelper } from './CommandHelper';

@Injectable({
  providedIn: 'root'
})
export class CheckResetCodeCommand {
  constructor(private httpClient: HttpClient, private commandHelper: CommandHelper) {
  }

  execute(resetCode: string, email: string): Observable<ServerMessage> {
    const url = '/api/checkResetCode';
    return this.commandHelper.getResponseStatusHandler(this.httpClient.post(url, { resetCode, email }), true);
  }
}

File: backend/src/Core.Command/Commands/CheckResetCode/CheckResetCodeCommand.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Text;
using Core.Common.Auth;
using Core.Common.Command;

namespace Core.Command.Commands.CheckResetCode
{
    public class CheckResetCodeCommand : CommandBase
    {
        [Required] public ResetCode ResetCode { get; }
        [Required] public string Email { get; }

        public CheckResetCodeCommand(ResetCode resetCode, string email)
        {
            ResetCode = resetCode;
            Email = email;
        }
    }
}